corehq/apps/reports/standard/deployments.py

```diff
@@ -352,11 +355,10 @@ def process_rows(self, users, fmt_for_export=False):
                 if last_build.get('app_id') and device and device.get('app_meta'):
                     device_app_meta = self.get_data_for_app(device.get('app_meta'), last_build.get('app_id'))
 
-            if last_sub and last_sub.get('commcare_version'):
-                commcare_version = _get_commcare_version(last_sub.get('commcare_version'))
-            else:
-                if device and device.get('commcare_version', None):
-                    commcare_version = _get_commcare_version(device['commcare_version'])
+            commcare_version, last_seen = get_commcare_version_and_date_from_last_usage(last_sub,
```

Comment:
It looks like the prior version of this function only set `last_seen` once, and that line still exists below (where its being set to `last_sub[submission_date]`. Are we setting `last_seen` here only to overwrite it? Should we ignore the `last_seen` return value from this function and rely on the other line? Or can we generalize this logic so that we remove the old line and replace it with the result of this function?

Reply:
Thanks for catching it. This is my oversight. 0d85d21970c4ab93bfffe1300cee233978fb8027
The logic cannot be generalized, because here the `last_seen` will only be last submission time, while the date returned by `get_commcare_version_and_date_from_last_usage` can be fall back to last_device.